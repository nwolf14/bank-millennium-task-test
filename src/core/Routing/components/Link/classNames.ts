export const className = 'Link';

export const classNames = {
  [className]: className,
};
