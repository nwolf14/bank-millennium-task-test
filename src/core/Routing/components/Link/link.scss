.Link {
  // to avoid error
  position: relative;
}
