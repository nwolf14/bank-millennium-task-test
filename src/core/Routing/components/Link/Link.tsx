import classnames from 'classnames';
import { FC, ReactNode, AnchorHTMLAttributes } from 'react';
import { Link as RouterLink } from 'react-router-dom';

// others
import { className as classNameLink, classNames } from './classNames';

// styles
import './link.scss';

export type TLinkProps = AnchorHTMLAttributes<HTMLAnchorElement> & {
  targetText?: '_blank' | '_self';
  children: ReactNode;
};

const Link: FC<TLinkProps> = ({
  children,
  target = '_self',
  targetText = 'Opens in a new window',
  className,
  href = '',
  ...props
}) => {
  const opensInNewWindow = target === '_blank';

  const isExternalLink = (href: string): boolean => {
    const externalLinkRegexp = /^https?:\/\//;
    return externalLinkRegexp.test(href);
  };

  if (isExternalLink(href) || opensInNewWindow) {
    return (
      <a
        href={href}
        target={target}
        className={classnames(classNames[classNameLink], className)}
        {...props}
      >
        {children}
        {opensInNewWindow && <span>{targetText}</span>}
      </a>
    );
  }

  return (
    <RouterLink to={href} className={classnames(className)} {...props}>
      {children}
    </RouterLink>
  );
};

export default Link;
