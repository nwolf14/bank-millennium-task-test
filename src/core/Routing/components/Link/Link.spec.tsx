import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

// components
import Link from './Link';

describe('Link', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Link href="/">Link</Link>
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Link', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Link href="https://">Link</Link>
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Link', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Link target="_blank">Link</Link>
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
