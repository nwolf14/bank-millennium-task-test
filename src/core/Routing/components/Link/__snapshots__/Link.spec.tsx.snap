// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link matches snapshot 1`] = `
<DocumentFragment>
  <a
    class=""
    href="/"
  >
    Link
  </a>
</DocumentFragment>
`;

exports[`Link matches snapshot 2`] = `
<DocumentFragment>
  <a
    class="Link"
    href="https://"
    target="_self"
  >
    Link
  </a>
</DocumentFragment>
`;

exports[`Link matches snapshot 3`] = `
<DocumentFragment>
  <a
    class="Link"
    href=""
    target="_blank"
  >
    Link
    <span>
      Opens in a new window
    </span>
  </a>
</DocumentFragment>
`;
